vizro-core/src/vizro/models/_components/graph.py

```diff
@@ -71,6 +75,39 @@ def __getitem__(self, arg_name: str):
             return self.type
         return self.figure[arg_name]
 
+    # Interaction methods
+    def _get_figure_interaction_input(self) -> Dict[str, State]:
+        """Required properties when using pre-defined `filter_interaction`."""
+        return {
+            "clickData": State(component_id=self.id, component_property="clickData"),
+            "modelID": State(component_id=self.id, component_property="id"),  # required, to determine triggered model
+        }
+
+    def _filter_interaction(
+        self, data_frame: pd.DataFrame, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
```

Comment:
```suggestion
        self, target: str, ctd_filter_interaction: Dict[str, CallbackTriggerDict]
```
Can we get rid of this and just access this model's data_frame with `data_manager._get_component_data(str(self.id))`?

Similarly if this model instance is always the `source_graph_id` then let's do that with `self.id` rather than going through `ctd_click_data["id"]`.

One of the advantages of having this code in the model is that we have direct access to `self` now, so if it's possible then we should use it wherever possible rather than passing the arguments in manually or extracting from Dash callback context.

The same comment also applies for the other models.

Reply:
I think this is not easily possible atm. 

We are already getting the data with `data_frame = data_manager._get_component_data(target)` in line 177 of `_action_utils.py`. This DF is then consecutively passed through filters.

If we were to load data afresh, we would skip the previous filters.

Besides, the triggered model provides the filtering information, the target component contains the to be filtered DF, so this would be the wrong DF anyway

Reply:
Ah ok, I see. Please could you add a quick comment saying what `data_frame` refers to here?